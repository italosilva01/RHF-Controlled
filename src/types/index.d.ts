export type Brand = {
  nome: string;
  codigo: number;
};

export type InputsForm = {
  brand: string;
  exampleRequired: string;
};
