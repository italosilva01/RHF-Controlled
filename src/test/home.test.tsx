import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Home from "../pages";
import { brands } from "@/mock";
import { Brand } from "@/types";
import { Form } from "react-hook-form";

const brandsValues = Object.values(brands) as unknown as Brand[];

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("Page Home", () => {
  test("renders without crashing", () => {
    const { container } = render(<Home brands={brandsValues} />);
    expect(
      screen.getByText("Consulte o valor de um veículo de forma gratuita")
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("render two initial autocompletes", () => {
    render(<Home brands={brandsValues} />);
    const inputs = screen.getAllByRole("combobox");
    expect(inputs).toHaveLength(2);
  });

  test("render two initial autocompletes with empty values", () => {
    render(<Home brands={brandsValues} />);
    const inputs = screen.getAllByRole("combobox");

    expect(inputs[0]).toHaveValue("");
    expect(inputs[1]).toHaveValue("");
  });

  test("Button submit form is disable", () => {
    render(<Home brands={brandsValues} />);
    const button = screen.getByTestId("button-submit");
    expect(button).toBeDisabled();
  });

  test("render the third autocompletes with empty values", async () => {
    render(<Home brands={brandsValues} />);

    const optionValue = "Acura";
    const inputs = screen.getAllByRole("combobox");

    // Simule a digitação do valor no input
    userEvent.click(inputs[0]);

    console.log("inputs[0]", inputs[0]);
    const option = await screen.findByText(optionValue);
    // const option = await screen.findByText((content, node) => {
    //   const hasText = (node) => node.textContent === optionValue;
    //   const nodeHasText = hasText(node);
    //   const childrenDontHaveText = Array.from(node?.children).every(
    //     (child) => !hasText(child)
    //   );

    //   return nodeHasText && childrenDontHaveText;
    // });
    // userEvent.type(inputs[0], optionValue);

    // // Aguarde o item aparecer na lista de opções
    // const listItem = await screen.findByText(optionValue);

    // Clique no item para selecioná-lo
    // userEvent.click(option);

    // // Verifique se o valor do input mudou
    // expect(inputs[0]).toHaveValue(optionValue);
  });

  // test("Form with Autocomplete using Portal", async () => {
  //   let rendered: ReturnType<typeof render>;

  //   await act(() => {
  //     rendered = render(<Form />, {
  //       wrapper: Wrapper,
  //       container: document.body,
  //     });
  //   });

  //   const { findByPlaceholderText, getByText } = rendered;

  //   const autocomplete = await findByPlaceholderText("Select Option");
  //   fireEvent.mouseDown(autocomplete);
  //   const { getByText: getByBodyText } = within(document.body);
  //   const option = getByBodyText("BuBuBu");
  //   expect(option).toBeInTheDocument();
  //   fireEvent.click(option as HTMLElement);

  //   expect(autocomplete).toHaveProperty("value", "BuBuBu");
  // });
});
