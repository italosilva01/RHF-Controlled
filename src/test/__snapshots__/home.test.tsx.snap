// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Home renders without crashing 1`] = `
<div>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-1kiosdo-MuiTypography-root"
  >
    Tabela Fipe
  </h4>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-1apd6s-MuiTypography-root"
  >
    Consulte o valor de um ve√≠culo de forma gratuita
  </h5>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-mnfu3y-MuiPaper-root-MuiCard-root"
  >
    <form>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1j0qgw8-MuiAutocomplete-root"
        name="brand"
        style="width: 450px;"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Marca
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wnwmuc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              placeholder="Escolha uma marca..."
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Marca
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1j0qgw8-MuiAutocomplete-root"
        name="model"
        style="width: 450px;"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r2:"
            id=":r2:-label"
          >
            Modelo
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wnwmuc-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              placeholder="Escolha um modelo..."
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Modelo
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1j0qgw8-MuiAutocomplete-root"
              name="year"
              style="width: 450px;"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1v4qvbo-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r4:"
                  id=":r4:-label"
                >
                  Ano
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-1wnwmuc-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r4:"
                    placeholder="Escolha um Ano..."
                    role="combobox"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-p1olib-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Ano
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-n2u3yb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-v35bdv-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Consultar pre√ßo
        </button>
      </div>
    </form>
  </div>
</div>
`;
